account/crypto.go

```diff
@@ -0,0 +1,235 @@
+package account
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"io"
+
+	"github.com/tendermint/go-crypto"
+	"golang.org/x/crypto/ed25519"
```

Comment:
No sure where we will end up with go-crypto... I think the aim will be to make it increasingly replaceable, but it may be that it does most of what we want. There's a few things it does I find a bit dubious, but we'll see...